ros/src/tl_detector/light_classification/tl_classifier.py

```diff
@@ -15,5 +85,75 @@ def get_classification(self, image):
             int: ID of traffic light color (specified in styx_msgs/TrafficLight)
 
         """
-        #TODO implement light color prediction
-        return TrafficLight.UNKNOWN
+        traffic_light_id = 4  # 4 equals to unknown
+
+        id_mapping = {4: TrafficLight.UNKNOWN,
+                      3: TrafficLight.GREEN,
+                      2: TrafficLight.YELLOW,
+                      1: TrafficLight.RED}
+
+        results = []
+        with self.session.graph.as_default():
+            boxes, scores, labels = self.session.run([self.detected_boxes_tensor,
+                                                      self.detected_scores_tensor,
+                                                      self.detected_labels_tensor],
+                                                     feed_dict={self.image_tensor: image})
+            # Filter for probability (score)
+            for i, score in enumerate(scores):
+                if score > self.detection_threshold:
+                    results.append({'box': boxes[i],
+                                    'score': score,
+                                    'id': labels[i]})
+
+	if len(results) > 0:
+	    # The boxes are encoded as xmin, xmax, ymin, ymax with normalized coordinates [0..1].
+	    # So lets find just the biggest box and take the traffic light state from it.
+	    max_sized_result = max(results, key=lambda bb: (bb['box'][1] - bb['box'][0]) * (bb['box'][3] - bb['box'][2]))
+	    traffic_light_id = max_sized_result['id']
```

Comment:
Inconsistent indentation here, line 108-112 should get indented another four spaces to the right.

Reply:
pushed a fix now.